src/metabase/query_processor/middleware/update_used_cards.clj

```diff
@@ -35,6 +47,8 @@
   (mu/fn [query :- ::qp.schema/query
           rff   :- ::qp.schema/rff]
     (letfn [(rff* [metadata]
-             (update-used-cards!* (set (lib.metadata/invoked-ids (qp.store/metadata-provider) :metadata/card)))
-             (rff metadata))]
+              (doseq [card-id (distinct (lib.metadata/invoked-ids (qp.store/metadata-provider) :metadata/card))]
+                (grouper/submit! @update-used-cards-queue {:id   card-id
+                                                           :timestamp (t/offset-date-time)}))
```

Comment:
There's a system call involved in getting the current time, so I'd put that in a let outside the loop.

Reply:
good call!